drivers/i2c/i2c_mcux.c

```diff
@@ -150,6 +152,7 @@ static int i2c_mcux_transfer(struct device *dev, struct i2c_msg *msgs,
 		 * successfully. e.g., nak, timeout, lost arbitration
 		 */
 		if (data->callback_status != kStatus_Success) {
+			I2C_MasterTransferAbort(base, &data->handle);
```

Comment:
... and this one. Please revisit your patch and reduce it to only include the actual bug fix.